src/components/[guild]/Requirements/components/ResetRequirementButton.tsx

```diff
@@ -1,25 +1,29 @@
-import { Button } from "@chakra-ui/react"
-import { useFormContext, useWatch } from "react-hook-form"
+import { Button, Spinner } from "@chakra-ui/react"
+import useEditRequirement from "components/create-guild/Requirements/hooks/useEditRequirement"
 import { useRequirementContext } from "./RequirementContext"
 
 const ResetRequirementButton = () => {
-  const { id } = useRequirementContext()
-  const { control, setValue } = useFormContext()
-  const requirements = useWatch({ name: "requirements", control })
-  const index = requirements?.findIndex((requirement) => requirement.id === id)
-
-  const onReset = () => {
-    setValue(`requirements.${index}.data.customName`, "", {
-      shouldDirty: true,
-    })
-    setValue(`requirements.${index}.data.customImage`, "", {
-      shouldDirty: true,
-    })
-  }
+  const requirement = useRequirementContext()
+  const { onSubmit, isLoading } = useEditRequirement(requirement.roleId)
 
   return (
-    <Button size={"sm"} onClick={onReset}>
-      Reset to original
+    <Button
+      size={"sm"}
+      isDisabled={isLoading}
+      onClick={() =>
+        onSubmit({
+          ...requirement,
+          data: { ...requirement.data, customName: "", customImage: "" },
+        })
+      }
+    >
+      {isLoading ? (
```

Comment:
We could just use the `Button` component's `isLoading` prop, that way we'd have a spinner by default & we wouldn't need to manually handle the `isDisabled` state ;)